httpx_ws/transport.py

```diff
@@ -39,26 +39,36 @@ def __init__(self, app: ASGIApp, scope: Scope) -> None:
         self._send_queue: queue.Queue[Message] = queue.Queue()
         self.connection = wsproto.WSConnection(wsproto.ConnectionType.SERVER)
         self.connection.initiate_upgrade_connection(scope["headers"], scope["path"])
+        self._exit_stack = contextlib.AsyncExitStack()
+        self._aentered = False
 
     async def __aenter__(
         self,
     ) -> tuple["ASGIWebSocketAsyncNetworkStream", bytes]:
-        self._task_group = await anyio.create_task_group().__aenter__()
-        self._task_group.start_soon(self._run)
+        if self._aentered:
+            raise RuntimeError(
+                "Cannot use ASGIWebSocketAsyncNetworkStream in a context manager twice"
+            )
+        self._aentered = True
+        async with contextlib.AsyncExitStack() as stack:
+            self._task_group = await stack.enter_async_context(anyio.create_task_group())
+            self._task_group.start_soon(self._run)
+
+            await self.send({"type": "websocket.connect"})
+            message = await self.receive()
 
-        await self.send({"type": "websocket.connect"})
-        message = await self.receive()
+            stack.push_async_callback(self.aclose)
 
-        if message["type"] == "websocket.close":
-            await self.aclose()
-            raise WebSocketDisconnect(message["code"], message.get("reason"))
+            if message["type"] == "websocket.close":
+                raise WebSocketDisconnect(message["code"], message.get("reason"))
```

Comment:
aclose was run here and then later in the `__aexit__`, I now run it once using a callback in the AsyncExitStack